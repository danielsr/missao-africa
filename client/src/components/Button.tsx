import React from 'react';

type ButtonProps = {
    label: string;
    onClick: Function;
};

function Button({ label, onClick }: ButtonProps) {
    return (
        <button
            className="border-0 p-2 bg-blue-900 hover:bg-blue-800 text-white rounded mr-2 text-sm"
            onClick={() => onClick()}
        >
            {label}
        </button>
    );
}

export default Button;
