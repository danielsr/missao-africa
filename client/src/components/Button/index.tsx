import React from 'react';
import classNames from 'classnames';

type ButtonProps = {
  label: string;
  onClick: Function;
  disabled?: boolean;
};

function Button({ label, onClick, disabled }: ButtonProps) {
  const className = classNames(
    'p-2 bg-blue-900 hover:bg-blue-800 text-white rounded mr-2 text-sm',
    {
      'opacity-75 cursor-not-allowed': disabled,
    }
  );

  return (
    <button className={className} onClick={() => onClick()} disabled={disabled}>
      {label}
    </button>
  );
}

export default Button;
