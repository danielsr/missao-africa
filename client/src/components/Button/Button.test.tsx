import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import Button from '.';

const fn = jest.fn();
const label = 'click';

afterEach(cleanup);

test('render label', () => {
  const { getByText } = render(<Button label={label} onClick={fn} />);
  const button = getByText(label);
  expect(button).toBeInTheDocument();
});

test('call onClick function on click', () => {
  const { getByText } = render(<Button label={label} onClick={fn} />);
  const button = getByText(label);
  fireEvent.click(button);
  expect(fn).toBeCalled();
});

test('render disabled class', () => {
  const { getByText } = render(<Button label={label} onClick={fn} disabled={true} />);
  const button = getByText(label);
  expect(button).toHaveClass('opacity-75 cursor-not-allowed');
});

test('do not call onClick function when disabled', () => {
  const functionToNotBeCalled = jest.fn();
  const { getByText } = render(
    <Button label={label} onClick={functionToNotBeCalled} disabled={true} />
  );
  const button = getByText(label);
  fireEvent.click(button);
  expect(functionToNotBeCalled).toBeCalledTimes(0);
});
