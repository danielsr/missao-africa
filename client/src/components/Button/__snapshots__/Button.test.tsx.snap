// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<DocumentFragment>
  <button
    class="p-2 bg-blue-900 hover:bg-blue-800 text-white rounded mr-2 text-sm"
  >
    click
  </button>
</DocumentFragment>
`;
